x-pack/plugins/security_solution/server/lib/siem_migrations/rules/task/agent/nodes/translate_query/prompt.ts

```diff
@@ -5,35 +5,84 @@
  * 2.0.
  */
 
+import type { RuleMigrationResources } from '../../../util/rule_resource_retriever';
 import type { MigrateRuleState } from '../../types';
 
-export const getEsqlTranslationPrompt = (
-  state: MigrateRuleState
-): string => `You are a helpful cybersecurity (SIEM) expert agent. Your task is to migrate "detection rules" from Splunk to Elastic Security.
-Below you will find Splunk rule information: the title, description and the SPL (Search Processing Language) query.
+export const getEsqlTranslationPrompt = (state: MigrateRuleState, query: string): string => {
+  return `You are a helpful cybersecurity (SIEM) expert agent. Your task is to migrate "detection rules" from Splunk to Elastic Security.
 Your goal is to translate the SPL query into an equivalent Elastic Security Query Language (ES|QL) query.
 
-Guidelines:
-- Start the translation process by analyzing the SPL query and identifying the key components.
+Splunk rule Information provided:
+- Below you will find Splunk rule information: the title (<<TITLE>>), the description (<<DESCRIPTION>>), and the SPL (Search Processing Language) query (<<SPL_QUERY>>).
+- Use all the information to analyze the intent of the rule, in order to translate into an equivalent ES|QL rule.
+- The fields in the Splunk query may not be the same as in the Elastic Common Schema (ECS), so you may need to map them accordingly.
+
+Translation process hints:
+- Analyze the SPL query and identify the key components.
+- Translate the SPL query into an equivalent ES|QL query using ECS (Elastic Common Schema) field names.
 - Always use logs* index pattern for the ES|QL translated query.
-- If, in the SPL query, you find a lookup list or macro that, based only on its name, you can not translate with confidence to ES|QL, mention it in the summary and
-add a placeholder in the query with the format [macro:<macro_name>(parameters)] or [lookup:<lookup_name>] including the [] keys, example: [macro:my_macro(first_param,second_param)] or [lookup:my_lookup].
 
-The output will be parsed and should contain:
+The output will be parsed and must contain:
 - First, the ES|QL query inside an \`\`\`esql code block.
 - At the end, the summary of the translation process followed in markdown, starting with "## Migration Summary".
 
-This is the Splunk rule information:
+Find the Splunk rule information below:
 
-<<SPLUNK_RULE_TITLE>>
+<<TITLE>>
 ${state.original_rule.title}
-<</SPLUNK_RULE_TITLE>>
+<</TITLE>>
 
-<<SPLUNK_RULE_DESCRIPTION>>
+<<DESCRIPTION>>
 ${state.original_rule.description}
-<</SPLUNK_RULE_DESCRIPTION>>
+<</DESCRIPTION>>
+
+<<SPL_QUERY>>
+${query}
+<</SPL_QUERY>>
+`;
+};
+
+export const getInlineQueryPrompt = (
+  state: MigrateRuleState,
+  resources: RuleMigrationResources
+): string => {
+  const resourcesContext = [];
+  if (resources.macro?.length) {
+    const macrosSummary = resources.macro
```

Comment:
super nit, you can make these into utils whether there's a common `generateSummary` util or macros and lists have their own...

Reply:
well, macros and lists have different summaries, I tried to keep the same format as they are in the spl queries.